Civi/Api4/Generic/Traits/ArrayQueryActionTrait.php

```diff
@@ -149,6 +150,14 @@ private function filterCompare($row, $condition) {
         $pattern = '/^' . str_replace('%', '.*', preg_quote($expected, '/')) . '$/i';
         return !preg_match($pattern, $value) == ($operator != 'LIKE');
 
+      case 'REGEXP':
+      case 'NOT REGEXP':
+        // MySQL uses case-insensitive REGEXP matching by default, do the same
+        // unless case-sensitivity is requested via the 'c' option
+        $modifiers = strpos($options, 'c') !== FALSE ? '' : 'i';
+        $pattern = '/' . str_replace('/', '\/', $expected) . '/' . $modifiers;
```

Comment:
I'm surprised that this works, since `\` characters tend to disappear in php strings. Usually you have to double-escape them, like 
```suggestion
        $pattern = '/' . str_replace('/', '\\/', $expected) . '/' . $modifiers;
```